### PR TITLE
guard needed for new conduit mesh bp

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

**Comment by cyrush** [comment]
@xjrc I think this simple guard works for both cases (old and new mesh bp). Does this make sense to you given the changes?

**Comment by xjrc** [comment]
@cyrush: Yes, this also looks like a valid solution to me. The logic is becoming a bit hard to follow so I hope we can move to the `blueprint::mesh::domains` solution eventually, but this should meet our needs for the time being.
